File: sdks/javascript/src/exceptions.ts
/**
 * Guardian API Exceptions
 */

export class GuardianAPIException extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'GuardianAPIException';
    Object.setPrototypeOf(this, GuardianAPIException.prototype);
  }
}

export class GuardianAPIError extends GuardianAPIException {
  public statusCode?: number;
  public response?: any;

  constructor(message: string, statusCode?: number, response?: any) {
    super(message);
    this.name = 'GuardianAPIError';
    this.statusCode = statusCode;
    this.response = response;
    Object.setPrototypeOf(this, GuardianAPIError.prototype);
  }
}

File: sdks/python/guardian_api/exceptions.py
"""
Guardian API Exceptions
"""


class GuardianAPIException(Exception):
    """Base exception for Guardian API client errors"""
    pass


class GuardianAPIError(GuardianAPIException):
    """Exception raised for API errors (4xx, 5xx responses)"""
    
    def __init__(self, message: str, status_code: int = None, response=None):
        super().__init__(message)
        self.status_code = status_code
        self.response = response

